### PR TITLE
Feature/api get lexicon name

### PR DESCRIPTION
**Pull request by AlejandroRG31**
Added a function to allow get all the lexicon names using the api methods 
Added a serializer of lexicon which has as values/variables: name source-language and dest-language
Added a test to check that this api method works correctly

### CODE REVIEW

**Comment by slamora** [comment]
While I was reviewing your code, I wonder if we could refactor `Lexicon` model to have a human code to identify it in a unique way and besides that keep a verbose name. Something like:
```python
class Lexicon(models.Model):
    code = models.SlugField(primary=True)
    name = models.CharField(max_length=32)
    description = models.TextField(blank=True)
    # Define languages using ISO 639-1 values
    src_language = models.CharField(max_length=2)
    dst_language = models.CharField(max_length=2)
```
**NOTE**: untested code .
**NOTE2** `code` slug may be generated automatically combining `src_language`and `dst_language` --> `{src}-{dst}`. E.g. `es-ar`

@AlejandroRG31 What do you think?

File: linguatec_lexicon/views.py [hunk 1]
```diff
@@ -80,6 +80,25 @@ class DefaultLimitOffsetPagination(LimitOffsetPagination):
...
+    def get_lexicon_names(self, request):
+        queryset = Lexicon.objects.all().order_by('name')
+        page = self.paginate_queryset(queryset)
+        if page is not None:
+            serializer = self.get_serializer(page, many=True)
+            return self.get_paginated_response(serializer.data)
+
+        serializer = self.get_serializer(queryset, many=True)
+        return Response(serializer.data)
+
```
Comment (slamora) [inline linguatec_lexicon/views.py]: I think that this method duplicates the response provided by default `list()` method.
If you need to customize queryset order_by you could override `get_queryset()` method.

File: tests/test_api.py [hunk 2]
```diff
@@ -29,6 +29,16 @@ def test_word_search_no_results(self):
         resp_json = resp.json()
         self.assertEqual(0, resp_json["count"])
 
+class LexiconAPITestCase(TestCase):
+    fixtures = ['lexicon-sample.json']
+
+    def test_lexicon(self):
+        resp = self.client.get('/api/lexicon/get_lexicon_names/')
```
Comment (slamora) [inline tests/test_api.py]: Standar API REST will expects that URL was `/api/lexicon/` OR better in plural `/api/lexicons/`.

**Review by slamora — CHANGES_REQUESTED** [review]

File: linguatec_lexicon/views.py (hunk 1 above)
Comment (AlejandroRG31) [reply linguatec_lexicon/views.py]: Yeah this method was like an overwritting of the default method, i have deleted this and refactored the default method

File: tests/test_api.py (hunk 2 above)
Comment (AlejandroRG31) [reply tests/test_api.py]: API refactored to use the default method and avoid using the overwritten one, also changed the default method name from lexicon to lexicons as you requested, which i also think is better and more intuitive

**Comment by AlejandroRG31** [comment]
> While I was reviewing your code, I wonder if we could refactor `Lexicon` model to have a human code to identify it in a unique way and besides that keep a verbose name. Something like:
> 
> ```python
> class Lexicon(models.Model):
>     code = models.SlugField(primary=True)
>     name = models.CharField(max_length=32)
>     description = models.TextField(blank=True)
>     # Define languages using ISO 639-1 values
>     src_language = models.CharField(max_length=2)
>     dst_language = models.CharField(max_length=2)
> ```
> 
> **NOTE**: untested code .
> **NOTE2** `code` slug may be generated automatically combining `src_language`and `dst_language` --> `{src}-{dst}`. E.g. `es-ar`
> 
> @AlejandroRG31 What do you think?

I think that a solution could be:

-Create a property or calculated value
-Make src_language  and dst_language unique together variables in order to make code also an unique variable
-Don't change name, in case we want to use name to save the name of the version, like "name = es-ar (version 1.1)"

```
@property
    def code(self):
        return (self.src_language + '-' + self.dst_language)

    name = models.CharField(unique=True, max_length=32)
   
    src_language = models.CharField(max_length=2)
    dst_language = models.CharField(max_length=2)

    unique_together = [['src_language', 'dst_language']]
```

The only problem is that code can not be used as a varible to filter, get .... (lexicon.object.get(code='ar-es')) but it can be used as a secondary name.

**Comment by slamora** [comment]
@AlejandroRG31 we need to be able to filter by Lexicon. So instead of implementing it as a `@property`  let's store it on the database.

Anyway, I agree with your proposal of `unique_together(src_language, dst_language)`.

**Review by slamora — APPROVED** [review]

**Comment by slamora** [comment]
@AlejandroRG31 I've seen that proposed changes are implemented on #10 . So, I guess that this PR could be merged and closed.

**Pull request merged**